plugins/dfu/dfu-target-stm.c

```diff
@@ -100,6 +100,9 @@ dfu_target_stm_upload_element (DfuTarget *target,
 	g_autoptr(GBytes) contents_truncated = NULL;
 	g_autoptr(GPtrArray) chunks = NULL;
 
+  /* this takes a long time on some devices */
+	dfu_device_set_timeout (device, 5000);
+
```

Comment:
Could we set this in dfu_target_stm_init() instead?

Reply:
yes, i can put it here